provider/pkg/provider/provider.go

```diff
@@ -1583,10 +1577,44 @@ func (k *kubeProvider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*p
 				newInputs.GetNamespace(), newInputs.GetName())
 		}
 
-		for k, v := range detailedDiff {
-			switch v.Kind {
-			case pulumirpc.PropertyDiff_ADD_REPLACE, pulumirpc.PropertyDiff_DELETE_REPLACE, pulumirpc.PropertyDiff_UPDATE_REPLACE:
-				replaces = append(replaces, k)
+		// Remove any ignored changes from the computed diff.
+		var ignorePaths []resource.PropertyPath
+		for _, ignore := range req.IgnoreChanges {
+			ignorePath, err := resource.ParsePropertyPath(ignore)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse ignore path: %w", err)
+			}
+			ignorePaths = append(ignorePaths, ignorePath)
+		}
+		var diffPaths []resource.PropertyPath
+		for p := range detailedDiff {
+			diffPath, err := resource.ParsePropertyPath(p)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse diff path: %w", err)
+			}
+			diffPaths = append(diffPaths, diffPath)
+		}
+		for _, ignorePath := range ignorePaths {
+			for _, diffPath := range diffPaths {
+				if ignorePath.Contains(diffPath) {
+					delete(detailedDiff, diffPath.String())
+					break
+				}
+			}
+		}
+
+		if len(detailedDiff) > 0 {
+			for k := range patchObj {
```

Comment:
I don't know this code very well but I wanted to ask if this change here is allright? This is now conditional. `detailedDiff` is now filtered by taking IgnoreChanges into account. The copying of patchObj to changes is now conditional. But it is not filtered. Is this OK? Consider:

- IgnoreChanges is meaningful
- detailedDiff got filtered, so `delete(detailedDiff, diffPath.String())` fired a few times
- remaining detailedDiff is non-empty; so `len(detailedDiff) > 0`
- now changes becomes a copy of `patchObj`, including changes that are filtered out of detailedDiff?

Reply:
Good points, thanks. I'll take another look at this logic today.